barretenberg/cpp/src/barretenberg/ultra_honk/ultra_composer.cpp

```diff
@@ -6,30 +6,91 @@
+
+    // Compute and store commitments to all precomputed polynomials
+    verification_key->q_m = commitment_key->commit(proving_key->q_m);
+    verification_key->q_l = commitment_key->commit(proving_key->q_l);
+    verification_key->q_r = commitment_key->commit(proving_key->q_r);
+    verification_key->q_o = commitment_key->commit(proving_key->q_o);
+    verification_key->q_c = commitment_key->commit(proving_key->q_c);
+    verification_key->sigma_1 = commitment_key->commit(proving_key->sigma_1);
+    verification_key->sigma_2 = commitment_key->commit(proving_key->sigma_2);
+    verification_key->sigma_3 = commitment_key->commit(proving_key->sigma_3);
+    verification_key->id_1 = commitment_key->commit(proving_key->id_1);
+    verification_key->id_2 = commitment_key->commit(proving_key->id_2);
+    verification_key->id_3 = commitment_key->commit(proving_key->id_3);
+    verification_key->lagrange_first = commitment_key->commit(proving_key->lagrange_first);
+    verification_key->lagrange_last = commitment_key->commit(proving_key->lagrange_last);
+
+    verification_key->q_4 = commitment_key->commit(proving_key->q_4);
+    verification_key->q_arith = commitment_key->commit(proving_key->q_arith);
+    verification_key->q_sort = commitment_key->commit(proving_key->q_sort);
+    verification_key->q_elliptic = commitment_key->commit(proving_key->q_elliptic);
+    verification_key->q_aux = commitment_key->commit(proving_key->q_aux);
+    verification_key->q_lookup = commitment_key->commit(proving_key->q_lookup);
+    verification_key->sigma_4 = commitment_key->commit(proving_key->sigma_4);
+    verification_key->id_4 = commitment_key->commit(proving_key->id_4);
+    verification_key->table_1 = commitment_key->commit(proving_key->table_1);
+    verification_key->table_2 = commitment_key->commit(proving_key->table_2);
+    verification_key->table_3 = commitment_key->commit(proving_key->table_3);
+    verification_key->table_4 = commitment_key->commit(proving_key->table_4);
+
+    // TODO(luke): Similar to the lagrange_first/last polynomials, we dont really need to commit to these polynomials
+    // due to their simple structure.
+    if constexpr (IsGoblinFlavor<Flavor>) {
+        verification_key->lagrange_ecc_op = commitment_key->commit(proving_key->lagrange_ecc_op);
+        verification_key->q_busread = commitment_key->commit(proving_key->q_busread);
+        verification_key->databus_id = commitment_key->commit(proving_key->databus_id);
+    }
+
+    instance->verification_key = std::move(verification_key);
+}
+
 template <UltraFlavor Flavor>
 std::shared_ptr<ProverInstance_<Flavor>> UltraComposer_<Flavor>::create_instance(CircuitBuilder& circuit)
 {
     circuit.add_gates_to_ensure_all_polys_are_non_zero();
     circuit.finalize_circuit();
     auto instance = std::make_shared<Instance>(circuit);
-    instance->commitment_key = compute_commitment_key(instance->proving_key->circuit_size);
+
+    if (!commitment_key) {
```

Comment:
Just curious if you saw a particular reason for this conditional. This is a pattern that has drifted around in the composer code for a while and I generally don't like it because it suggests some uncertainty about when/whether or not things are computed. If it's needed for some reason, no big deal. Otherwise, I say we just compute it.